site/app/controllers/admin/PlagiarismController.php

```diff
@@ -533,161 +636,280 @@ public function saveNewPlagiarismConfiguration(string $new_or_edit, string $grad
             $return_url = $this->core->buildCourseUrl(['plagiarism', 'configuration', 'edit']) . "?gradeable_id={$gradeable_id}&config_id={$config_id}";
         }
 
-        // Check for invalid gradeable/config IDs
         try {
-            $this->verifyGradeableAndConfigAreValid($gradeable_id, $config_id);
+            $config_id = intval($config_id);
         }
         catch (Exception $e) {
-            $this->core->addErrorMessage($e->getMessage());
+            $this->core->addErrorMessage("Error: Config ID must be a valid integer configuration ID");
             return new RedirectResponse($return_url);
         }
 
+        // Check if Lichen job is already running
+        if (file_exists($this->getQueuePath($gradeable_id, $config_id)) || file_exists($this->getProcessingQueuePath($gradeable_id, $config_id))) {
+            $this->core->addErrorMessage("A job is already running for the gradeable. Try again after a while.");
+            return new RedirectResponse($return_url);
+        }
 
-        if ($config_id !== null) {
-            // Check if Lichen job is already running
-            if (file_exists($this->getQueuePath($gradeable_id, $config_id)) || file_exists($this->getProcessingQueuePath($gradeable_id, $config_id))) {
-                $this->core->addErrorMessage("A job is already running for the gradeable. Try again after a while.");
+        // Generate a unique number for this version of the gradeable //////////
+        if ($new_or_edit === "new") {
+            try {
+                $config_id = $em->getRepository(PlagiarismConfig::class)
+                    ->findOneBy(["gradeable_id" => $gradeable_id], ["config_id" => "DESC"]);
+                if ($config_id === null) {
+                    $config_id = 1;
+                }
+                else {
+                    $config_id = $config_id->getConfigID() + 1;
+                }
```

Comment:
What is the relation of this `$config_id` versus the one that gets passed into this function?

Reply:
This particular piece of code is something which has been bothering me for some time but I'm not sure how to do it better.  If this is an edit to an existing config, a `config_id` will be passed to the function.  If this is a new config, we need to create a new `config_id` which is what this code does.